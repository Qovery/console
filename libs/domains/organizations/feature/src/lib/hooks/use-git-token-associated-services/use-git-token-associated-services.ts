import { useQuery } from '@tanstack/react-query'
import { queries } from '@qovery/state/util-queries'

export interface UseGitTokenAssociatedServicesProps {
  organizationId: string
  gitTokenId: string
}

export function useGitTokenAssociatedServices({ organizationId, gitTokenId }: UseGitTokenAssociatedServicesProps) {
  return useQuery({
    ...queries.organizations.gitTokenAssociatedServices({ organizationId, gitTokenId }),
  })
}

export default useGitTokenAssociatedServices
