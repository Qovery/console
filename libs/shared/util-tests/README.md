# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
