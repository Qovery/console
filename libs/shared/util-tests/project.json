{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
