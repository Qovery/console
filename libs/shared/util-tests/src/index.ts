export * from './lib/shared-util-tests'
