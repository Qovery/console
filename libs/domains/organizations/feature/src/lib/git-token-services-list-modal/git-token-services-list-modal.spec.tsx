import { GitTokenAssociatedServiceType } from 'qovery-typescript-axios'
import { renderWithProviders, screen } from '@qovery/shared/util-tests'
import * as useGitTokenAssociatedServices from '../hooks/use-git-token-associated-services/use-git-token-associated-services'
import GitTokenServicesListModal, {
  type GitTokenServicesListModalProps,
  groupByProjectEnvironmentsServices,
} from './git-token-services-list-modal'

const useGitTokenAssociatedServicesMockSpy = jest.spyOn(
  useGitTokenAssociatedServices,
  'useGitTokenAssociatedServices'
) as jest.Mock

const props: GitTokenServicesListModalProps = {
  organizationId: '0000-0000-0000',
  gitTokenId: '0000-0000-0000',
  onClose: jest.fn(),
  associatedServicesCount: 3,
}

const data = [
  {
    project_id: '1',
    project_name: 'Project 1',
    environment_id: '1',
    environment_name: 'Development',
    service_id: '101',
    service_name: 'Service 1',
    service_type: GitTokenAssociatedServiceType.APPLICATION,
  },
  {
    project_id: '1',
    project_name: 'Project 1',
    environment_id: '1',
    environment_name: 'Development',
    service_id: '102',
    service_name: 'Service 2',
    service_type: GitTokenAssociatedServiceType.HELM,
  },
  {
    project_id: '2',
    project_name: 'Project 2',
    environment_id: '1',
    environment_name: 'Staging',
    service_id: '201',
    service_name: 'Service 3',
    service_type: GitTokenAssociatedServiceType.CRON,
  },
]

describe('GitTokenServicesListModal', () => {
  beforeEach(() => {
    useGitTokenAssociatedServicesMockSpy.mockReturnValue({
      data,
    })
  })

  it('should render successfully', () => {
    const { baseElement } = renderWithProviders(<GitTokenServicesListModal {...props} />)
    expect(baseElement).toBeTruthy()
  })

  it('should group data by projects, environments, and services correctly', () => {
    const result = groupByProjectEnvironmentsServices(data)

    expect(result).toHaveLength(2)

    expect(result[0].project_id).toBe('1')
    expect(result[0].project_name).toBe('Project 1')
    expect(result[0].environments).toHaveLength(1)
    expect(result[0].environments[0].environment_id).toBe('1')
    expect(result[0].environments[0].environment_name).toBe('Development')
    expect(result[0].environments[0].services).toHaveLength(2)

    expect(result[1].project_id).toBe('2')
    expect(result[1].project_name).toBe('Project 2')
    expect(result[1].environments).toHaveLength(1)
    expect(result[1].environments[0].environment_id).toBe('1')
    expect(result[1].environments[0].environment_name).toBe('Staging')
    expect(result[1].environments[0].services).toHaveLength(1)
  })

  it('should match snapshots', async () => {
    const { baseElement, userEvent } = renderWithProviders(<GitTokenServicesListModal {...props} />)

    const triggers = screen.getAllByRole('button')
    screen.getByText(/project 1/i)
    await userEvent.click(triggers[0])

    const triggerEnvironment = screen.getByText(/development/i).parentElement
    await userEvent.click(triggerEnvironment!)

    expect(baseElement).toMatchSnapshot()
  })
})
