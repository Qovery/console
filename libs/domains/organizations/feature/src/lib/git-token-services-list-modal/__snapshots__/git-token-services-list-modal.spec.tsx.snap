// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitTokenServicesListModal should match snapshots 1`] = `
<body>
  <div>
    <section
      class="flex flex-col p-6"
    >
      <h1
        class="font-bold text-neutral-400 mb-6 text-2xl"
      >
        Associated services (
        3
        )
      </h1>
      <div
        class="relative w-full mb-3"
      >
        <span
          class="icon-solid-magnifying-glass shrink-0 absolute left-3 top-1/2 -translate-y-1/2 block text-base text-neutral-400 dark:text-white leading-none"
          role="img"
        />
        <input
          class="w-full rounded border border-neutral-250 dark:border-neutral-350 dark:bg-transparent text-neutral-400 dark:text-white placeholder:text-neutral-350 dark:placeholder:text-neutral-250 pl-10 pr-6 focus:outline-none focus:border-brand-400 focus:transition-[border-color] h-9 text-xs"
          data-testid="input-search"
          name="search"
          placeholder="Search by project, environment, service name"
          type="text"
        />
      </div>
      <div
        class="border border-neutral-250 bg-neutral-100 py-2 px-4 rounded"
        data-orientation="vertical"
      >
        <div
          class="mt-px overflow-hidden first:mt-0 first:rounded-t last:rounded-b"
          data-orientation="vertical"
          data-state="open"
        >
          <div
            class="flex items-center w-full pr-5 py-2 text-sm outline-none text-neutral-400"
          >
            <button
              aria-controls="radix-:rh:"
              aria-expanded="true"
              class="cursor-pointer inline-flex items-center justify-center border border-neutral-250 text-3xs w-4 h-4 rounded mr-5"
              data-orientation="vertical"
              data-radix-collection-item=""
              data-state="open"
              id="radix-:rg:"
              type="button"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-minus "
              />
            </button>
            Project 1
          </div>
          <div
            aria-labelledby="radix-:rg:"
            class="data-[state=open]:animate-slidein-down-sm-faded data-[state=closed]:slidein-up-sm-faded ml-2 pl-[26px] overflow-hidden border-l border-neutral-200"
            data-orientation="vertical"
            data-state="open"
            id="radix-:rh:"
            role="region"
            style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
          >
            <div
              data-orientation="vertical"
            >
              <div
                class="mt-px overflow-hidden first:mt-0 first:rounded-t last:rounded-b"
                data-orientation="vertical"
                data-state="closed"
              >
                <div
                  class="flex items-center w-full pr-5 py-2 text-sm outline-none text-neutral-400"
                >
                  <button
                    aria-controls="radix-:rl:"
                    aria-expanded="false"
                    class="cursor-pointer inline-flex items-center justify-center border border-neutral-250 text-3xs w-4 h-4 rounded mr-5"
                    data-orientation="vertical"
                    data-radix-collection-item=""
                    data-state="closed"
                    id="radix-:rk:"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="fa-solid fa-plus "
                    />
                  </button>
                  <a
                    class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-brand-500 hover:text-brand-600 text-sm"
                    href="/organization/0000-0000-0000/project/1/environment/1/services/general"
                  >
                    Development
                  </a>
                </div>
                <div
                  aria-labelledby="radix-:rk:"
                  class="data-[state=open]:animate-slidein-down-sm-faded data-[state=closed]:slidein-up-sm-faded ml-2 pl-[26px] overflow-hidden border-l border-neutral-200"
                  data-orientation="vertical"
                  data-state="closed"
                  hidden=""
                  id="radix-:rl:"
                  role="region"
                  style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="mt-px overflow-hidden first:mt-0 first:rounded-t last:rounded-b"
          data-orientation="vertical"
          data-state="closed"
        >
          <div
            class="flex items-center w-full pr-5 py-2 text-sm outline-none text-neutral-400"
          >
            <button
              aria-controls="radix-:rj:"
              aria-expanded="false"
              class="cursor-pointer inline-flex items-center justify-center border border-neutral-250 text-3xs w-4 h-4 rounded mr-5"
              data-orientation="vertical"
              data-radix-collection-item=""
              data-state="closed"
              id="radix-:ri:"
              type="button"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-plus "
              />
            </button>
            Project 2
          </div>
          <div
            aria-labelledby="radix-:ri:"
            class="data-[state=open]:animate-slidein-down-sm-faded data-[state=closed]:slidein-up-sm-faded ml-2 pl-[26px] overflow-hidden border-l border-neutral-200"
            data-orientation="vertical"
            data-state="closed"
            hidden=""
            id="radix-:rj:"
            role="region"
            style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
          />
        </div>
      </div>
    </section>
  </div>
</body>
`;
